tests/dir_helpers.py

```diff
@@ -285,3 +293,21 @@ def erepo_dir(tmp_path_factory, monkeypatch):
         path.dvc.close()
 
     return path
+
+
+@pytest.fixture
+def erepo_dir_no_remote_config(tmp_path_factory, monkeypatch):
+    from dvc.repo import Repo
+
+    path = TmpDir(fspath_py35(tmp_path_factory.mktemp("erepo")))
+
+    # Chdir for git and dvc to work locally
+    with path.chdir():
+        _git_init()
+        path.dvc = Repo.init()
+        path.scm = path.dvc.scm
+        path.scm.commit("init dvc")
+
+        path.dvc.close()
+
+    return path
```

Comment:
Unfortunately removing that `config` part still won't help, as https://github.com/iterative/dvc/blob/master/dvc/external_repo.py#L91 is still in place, which automatically sets it back when cloning 🙁

How about we simply mock `cloud.pull` to raise NoRemoteError in our test?

Reply:
I am bit confused here. When I execute:
```
python -m tests tests/func/test_import.py::test_import_cached_file
```
I do see a config file under location `/tmp/pytest-of-sujith/pytest-0/popen-gw0/erepo0/.dvc/config` in my machine. The file is actually empty. So while cloning it sets back but the content is empty.. Kindly correct me if my obeservation is wrong.

Reply:
@sharidasan It creates `.dvc/config.local` though, not `.dvc/config`.